packages/compass-aggregations/src/modules/pipeline-builder/stage-editor.ts

```diff
@@ -453,23 +507,49 @@ export const addStage = (
   after?: number
 ): PipelineBuilderThunkAction<void, StageAddAction> => {
   return (dispatch, getState, { pipelineBuilder }) => {
-    const stage = pipelineBuilder.addStage(after);
+    const {
+      pipelineBuilder: {
+        stageEditor: { stages },
+      },
+    } = getState();
+    const addAfter = after ?? stages.length;
+    const wizardsUntilAfter = stages
+      .slice(0, addAfter + 1)
+      .filter(({ type }) => type === 'wizard');
+    const addAfterIdxInPipeline = addAfter - wizardsUntilAfter.length;
+
+    const stage = pipelineBuilder.addStage(addAfterIdxInPipeline);
     track('Aggregation Edited', {
       num_stages: getState().pipelineBuilder.stageEditor.stages.length,
```

Comment:
The `num_stages` needs to filter out the wizards now. We can create a helper and reuse it across

Reply:
Took care of all the tracking that needs updated.